jina/executors/indexers/vector/numpy.py

```diff
@@ -138,6 +134,20 @@ def query(self, keys: np.ndarray, top_k: int, *args, **kwargs) -> Tuple['np.ndar
         dist = np.take_along_axis(dist, idx, axis=1)
         return self.int2ext_key[idx], dist
 
+    def _load_numpy(self, abspath):
+        if not path.exists(abspath):
+            self.logger.warning('numpy data not found: {}'.format(abspath))
+            return None
+        result = None
+        try:
+            if self.num_dim and self.dtype:
+                with gzip.open(abspath, 'rb') as fp:
+                    result = np.frombuffer(fp.read(), dtype=self.dtype).reshape([-1, self.num_dim])
+        except EOFError:
+            self.logger.error(
+                f'{self.index_abspath} is broken/incomplete, perhaps forgot to ".close()" in the last usage?')
+        return result
+
```

Comment:
wait, are you sure to make this refactoring just for a helper function to load training data in faiss. afaik, the faiss training data loading is just a quick temporal solution. That helper function won't last long.

In fact, you don't need to load from a gzipped numpy for the training data. A simple numpy array would be good, simply use `np.load`, it is less error-prone and more generic. Otherwise, how do you expect user to generate `train.gz` in the first place??

Reply:
The training data loading is necessary for faiss. Therefore, I consider two scenarios.
1. the user can use indexed data for training, which is kept in `index.tgz` with the `NumpyIndexer`.
2. as you suggest, the user can simple feed a numpy array. This is the reason why we have the wrapper function `_load_training_data`.